src/tab/types.ts

```diff
@@ -0,0 +1,31 @@
+import { Dispatch, SetStateAction } from "react";
+
+export type TabItemRenderMode = "default" | "controlled";
+
+export interface TabProps {
+    children: JSX.Element | JSX.Element[];
+    /**
+     * Specifies the render mode of the tab items.
+     *
+     * Specify `"controlled"` if you want to control the
+     * rendering of each time item.
+     *
+     * Leave undefined or specify `"default"` if you want
+     * the `Tab` component to control instead.
+     */
+    tabItemRenderMode?: TabItemRenderMode | undefined;
+    currentActive?: number;
```

Comment:
Yea so it's exactly like you said because `currentActive` was also to handle initial visible. I think we can go with the `initialActive` approach and i'll remove the `tabItemRenderMode`

Reply:
I will however use `onTabClick` to determine if it is controlled or not.